alot/db/folder.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+
+from alot.utils import imap_utf7
+
+
+class Folder(object):
```

Comment:
I really doubt we need such a class and could instead do with strings representing paths relative to $NOTMUCH_ROOT. The difference to threads or messages, for which we do have such wrappers, is that those are concepts that directly appear in libnotmuch. Folders appear just as part of the query syntax for finding messages/threads. I think we should simply hand the `FolderWidget` that appears in the folder tree a reference to the database manager, so that it can query and display the result of `notmuch count folder:$MYPATH`. Keep it simple.